silkworm/silkrpc/http/request_parser.cpp

```diff
@@ -24,270 +24,80 @@
+    if (req.content_length != 0 && req.content.length() < req.content_length) {
+        for (size_t ii = 0; ii < slen; ii++) {
+            req.content.push_back(begin[ii]);
+        }
+        if (req.content.length() < req.content_length)
+            return ResultType::indeterminate;
+        else
+            return ResultType::good;
+    }
+
+    num_headers = sizeof(headers) / sizeof(headers[0]);
+    auto res = phr_parse_request(begin, slen, &method_name, &method_len, &path, &path_len, &minor_version, headers, &num_headers, last_len);
+    if (res < 0) {
+        return ResultType::bad;
+    }
+
+    bool expect_request = false;
+    bool content_length_present = false;
+
+    for (size_t ii = 0; ii < num_headers; ii++) {
+        if (memcmp(headers[ii].name, "Content-Length", headers[ii].name_len) == 0) {
+            req.content_length = static_cast<uint32_t>(atoi(headers[ii].value));
+            content_length_present = true;
+        }
+
+        else if (memcmp(headers[ii].name, "Expect", headers[ii].name_len) == 0) {
+            expect_request = true;
+        }
+
+        else if (memcmp(headers[ii].name, "Authorization", headers[ii].name_len) == 0) {
+            req.headers.emplace_back();
+            for (size_t index = 0; index < static_cast<size_t>(headers[ii].name_len); index++) {
+                req.headers.back().name.push_back(headers[ii].name[index]);
             }
-        case content_start:
-            req.content.push_back(input);
-            if (req.content.length() < req.content_length) {
-                return ResultType::indeterminate;
-            } else {
-                return ResultType::good;
+            for (size_t index = 0; index < static_cast<size_t>(headers[ii].value_len); index++) {
+                req.headers.back().value.push_back(headers[ii].value[index]);
             }
+        }
     }
-    return ResultType::bad;
-}
-
-inline bool RequestParser::is_char(int c) {
-    return c >= 0 && c <= 127;
-}
 
-inline bool RequestParser::is_ctl(int c) {
-    return (c >= 0 && c <= 31) || (c == 127);
-}
+    if (content_length_present == false) {
+        return ResultType::bad;
+    }
 
-inline bool RequestParser::is_tspecial(int c) {
-    switch (c) {
-        case '(':
-        case ')':
-        case '<':
-        case '>':
-        case '@':
-        case ',':
-        case ';':
-        case ':':
-        case '\\':
-        case '"':
-        case '/':
-        case '[':
-        case ']':
-        case '?':
-        case '=':
-        case '{':
-        case '}':
-        case ' ':
-        case '\t':
-            return true;
-        default:
-            return false;
+    if (req.content_length == 0) {
+        return ResultType::good;
     }
-}
 
-inline bool RequestParser::is_digit(int c) {
-    return c >= '0' && c <= '9';
+    req.content.resize(slen - static_cast<size_t>(res));
+    memcpy(req.content.data(), &begin[res], slen - static_cast<size_t>(res));
```

Comment:
Either use `std::memcpy` here and below or add `using namespace std;` at the top

Reply:
ok